src/app/components/search/search.component.ts

```diff
@@ -0,0 +1,35 @@
+// search.component.ts
+
+import { Component, OnInit } from '@angular/core';
+import { JokeService } from '../../joke.service';
+
+@Component({
+  selector: 'app-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.css']
+})
+export class SearchComponent implements OnInit {
+
+  searchValue: string;
+
+  constructor(private jokeService: JokeService) { }
+
+  // get search results array from jokeservice
+  get searchData(): any {
+    return this.jokeService.searchList;
+  }
+
+  searchButton (string) {
```

Comment:
Method names should always be a verb, not a noun. It helps others to identify the action that a method is intended to perform. Descriptive comments for each method would also be very helpful.